### ISSUE TITLE
Matrix collinearity

### ISSUE THREAD

**Issue by Marsoa** [zhangxiang0822/ShiftShareSEStata#6]
Good morning Xiang,

I am trying to use the ivreg_ss and I get the error message "You have collinear share variables (Share matrix has colinear columns)".  
I am working with international trade data, and the shares that I am using are the weight of each product in a given country's export basket (that is, the composition of a country's export basket), which by definition add up to 1. I thought this was the problem, and I have modified the shift-share variable to have the weight of each exported product in country's GDP instead of in country's exports, which is not equal to 1, but I get the same error message. In fact, I get this message for every  one of the shares, out of around 5000 (we work with highly disaggregated trade data): "omitted because of collinearity". 
I am not sure why do I still come across this error.  I though that by rescaling product shares on country's GDP I was getting rid of this problem, in a similar way as you propose in the "help ivreg_ss" section in Stata.

I hope I have been clear in stating the problem. Could you please provide any indication?
Thank you very much.

Kind regards,
Marta

**Comment by rosasanchisguarner** [comment]
Hello! I am having exactly the same problem, it says all my shares are collinear. Did you manage to fix it?
Thanks
Rosa

**Comment by Marsoa** [comment]
> Hello! I am having exactly the same problem, it says all my shares are collinear. Did you manage to fix it? Thanks Rosa

Hi Rosa,
In my case, the issue was that the number of shifters (5000 products) is higher than the number of observations (96 countries). The inference method from Adao et al. does not work in this case.

Hope that this helps,
Marta

**Comment by giuliaromani1** [comment]
Hi!

I also get a similar message stating: "You have collinear share variables (Share matrix has collinear columns)."

My setting is a traditional one, where the endogenous variable is LLM employment, instrumented with a Bartik shift-share. The shares are based on past industry shares in the LLM, and the shifts are the increases in national employment in a given industry.

I estimate an equation including observations for the start (2005) and end (2019) periods, along with LLM fixed effects. This is equivalent to first-differencing the regression variables, so it's essentially a long-difference estimation. I use past industry shares from 1995.

Given how I build the estimation, I have two time periods, both associated with the same LLM past shares, as they are fixed to 1995. Do you think this could be the source of the collinearity problem?

Thanks for your help!
Best,
Giulia

**Comment by kolesarm** [comment]
@Marsoa That's right, the inference method requires the number of shifters to be smaller than the number of observations. One solution is to aggregate the shifters.

Note the latest version of the package (here on github) automatically handles collinear shares if the collinearity just comes from a few shifters having the same share pattern.

@giuliaromani1 Are you implementing the panel data as we outline in Section 5.2 of our paper, in particulare the definition of shares in eq (41)? (see also our empirical application to ADH and the replication code for it) Are you getting the same error if you estimate it in first differences?
